spec.md

```diff
@@ -417,22 +417,30 @@ how Platforms might expose these values to their users.
 
 | Response Field | Type | Description |
 | --- | --- | --- |
-| [create](#input-parameters-object) | object | The schema definition for creating a Service Instance. |
-| [update](#input-parameters-object) | object | The schema definition for updating a Service Instance. |
+| [create](#json-schema-object) | object | The schema definition for creating a Service Instance. |
+| [update](#json-schema-object) | object | The schema definition for updating a Service Instance. |
 
 ##### Service Binding Object
 
 | Response Field | Type | Description |
 | --- | --- | --- |
-| [create](#input-parameters-object) | object | The schema definition for creating a Service Binding. |
+| [create](#json-schema-object) | object | The schema definition for creating a Service Binding. |
 
-##### Input Parameters Object
+
+##### JSON Schema Object
```

Comment:
This needs to be called something else. In my previous comment I suggested JSON Schema**s** Object, but I now see that the 's' might be too easy to miss. 

It definitely can't be called "JSON Schema Object", because if you look in the table under this heading, you'll see the Type column refer to "JSON schema object" which makes it appear as though there's a recursion there.

A few ideas for the name (but none of them fits perfectly):
- InputOutputSchema (or InputOutputSchemas)
- RequestResponseSchema (or RequestResponseSchemas)
- JSONSchema**s**

Reply:
This is a good catch. It does make it unclear if this is defining the form for Json Schema Object (not what I wanted) or something else. I renamed it from Input Parameters Object to not restrict it to just inout parameters...

Maybe:

* Schemas Definition Object
* Expected Schemas Object
* Defined Schemas
* Provided Schemas (or Broker Provided Schemas?)

Reply:
Looking at this, I think there are actually two different types here:

- The type for service instances, which should allow the broker to report parameter schemas for create and update
- The type for bindings, which should allow the broker to report parameter schema for create and an output schema for credentials

Seems like we ought to differentiate the types as:

- Service instance schema object
- Service binding schema object

WDYT?

Reply:
I agree there are two different uses for the objects defined, but they are the same type of object. There is nothing special about the JSON Schema object, the special part has to do with how the platform interprets it (as the form of an expected input or expected output). Regardless of usage, the JSON Schema definition is the same.

Reply:
How do people feel about something along the lines of using "dialect", used to mean the custom way this service will be communicating? @pmorie @luksa